web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount.utils.tsx

```diff
@@ -1,37 +1,171 @@
-import {
-  CreditDetails,
-  CreditsVintages,
-} from 'web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.types';
+import { CardSellOrder } from 'web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.types';
 
-import { Currency } from 'web-components/src/components/DenomIconWithCurrency/DenomIconWithCurrency.constants';
+import { denomToMicro, microToDenom } from 'lib/denom.utils';
 
-export const getCurrencyPrice = (
-  currency: Currency,
-  creditDetails: CreditDetails[],
-) => {
-  return (
-    creditDetails.find(credit => credit.currency === currency)?.creditPrice || 1
-  );
-};
+import { PAYMENT_OPTIONS } from 'pages/BuyCredits/BuyCredits.constants';
+import { PaymentOptionsType } from 'pages/BuyCredits/BuyCredits.types';
+import { UISellOrderInfo } from 'pages/Projects/AllProjects/AllProjects.types';
+
+import { CURRENCY_AMOUNT } from './CreditsAmount.constants';
 
 export const getCreditsAvailablePerCurrency = (
-  currency: Currency,
-  creditDetails: CreditDetails[],
+  paymentOption: PaymentOptionsType,
+  filteredCryptoSellOrders: Array<UISellOrderInfo> | undefined,
+  cardSellOrders: Array<CardSellOrder>,
+  creditTypePrecision?: number | null,
 ) => {
-  return (
-    creditDetails.find(credit => credit.currency === currency)
-      ?.availableCredits || 0
-  );
+  return paymentOption === PAYMENT_OPTIONS.CARD
+    ? cardSellOrders.reduce((prev, cur) => prev + Number(cur.quantity), 0)
+    : filteredCryptoSellOrders?.reduce((prev, cur) => {
+        return parseFloat(
+          (prev + Number(cur.quantity)).toFixed(creditTypePrecision || 6),
+        );
+      }, 0) || 0;
 };
 
-export const getVintageCredits = (
-  creditVintageOptions: string[],
-  creditVintages: CreditsVintages[],
-) => {
-  return creditVintageOptions.reduce((sum: number, option: string) => {
-    const credits =
-      creditVintages.find(vintage => vintage.batchDenom === option)?.credits ||
-      '0';
-    return sum + +credits;
-  }, 0);
+export function getSpendingCap(
+  paymentOption: PaymentOptionsType,
+  filteredCryptoSellOrders: Array<UISellOrderInfo> | undefined,
+  cardSellOrders: Array<CardSellOrder>,
+) {
+  return paymentOption === PAYMENT_OPTIONS.CARD
+    ? cardSellOrders.reduce(
+        (prev, cur) => prev + Number(cur.quantity) * cur.usdPrice,
+        0,
+      )
+    : microToDenom(
+        filteredCryptoSellOrders?.reduce(
+          (prev, cur) => prev + Number(cur.quantity) * Number(cur.askAmount),
+          0,
+        ) || 0,
+      );
+}
+
+type GetCreditsAmountParams = {
+  value: number;
+  card: boolean;
+  orderedSellOrders: UISellOrderInfo[];
+  creditTypePrecision?: number | null;
+};
+export const getCreditsAmount = ({
```

Comment:
@r41ph just wanted to mention that those functions `getCreditsAmount ` and `getCurrencyAmount` are one of the most important stuff to review in this PR, since it infers the sell orders (by the cheapest) based on the credits or currency amount provided

Reply:
@blushi this looks great! I've tested it and it seems to work as expected. Just a few minor suggestions.